### PR TITLE
[bitnami/common] Add app and version to common.labels.standard

### PR DESCRIPTION
**Pull request by Starefossen**
**Description of the change**

Add `app` and `version` labels that are comonly used by Istio.

**Benefits**

This will properly group and display the application in Istio (and other tools).

**Possible drawbacks**

A few bytes overhead

**Applicable issues**

n/a

**Additional information**

* https://istio.io/latest/docs/ops/deployment/requirements/

> Pods with app and version labels: We recommend adding an explicit app label and version label to the specification of the pods deployed using a Kubernetes Deployment. The app and version labels add contextual information to the metrics and telemetry that Istio collects.
>
>  * The app label: Each deployment should have a distinct app label with a meaningful value. The app label is used to add contextual information in distributed tracing.
>
> * The version label: This label indicates the version of the application corresponding to the particular deployment.

**Checklist** 
- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/common/Chart.yaml
Comment (juan131) [inline bitnami/common/Chart.yaml:5]: Please update the version too.

File: bitnami/common/templates/_labels.tpl
Comment (juan131) [inline bitnami/common/templates/_labels.tpl:6]: We already included this label for certain charts to be compatible with Istio (see https://github.com/bitnami/charts/pull/2286)
That said, I'd like Istio to move forward to embrace the [default k8s recommended labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/). `app`, `version`, etc. labels are not recommended by k8s since quite a lot time.
I have mixed feelings about adding this since we're duplicating labels.

File: bitnami/common/templates/_labels.tpl
Comment (juan131) [inline bitnami/common/templates/_labels.tpl:11]: We should be extremely cautious when including labels that change everytime the version is bumped (`appVersion` in this case). In the past, this lead into upgradability issues.

**Review by juan131 — CHANGES_REQUESTED** [review]
@javsalgar @carrodher I'd like you to share your thoughts on this.

File: bitnami/common/templates/_labels.tpl
Comment (carrodher) [reply bitnami/common/templates/_labels.tpl:6]: My 2cts is to follow the k8s recommendations as much as possible, if there is a specific asset that needs to follow a different approach due to a widely common use case we can evaluate an exception for this kind of case. But adding non-recommended and duplicated labels to the different k8s objects in this library chart that is going to be applied to the whole catalog doesn't sound like the best option to me.

Do you know why Istio is not using the recommended labels (any open issue or similar)? I'm just thinking out loud, but I am not sure if we should create this (potentially risky) workaround to support something it's not recommended by the core maintainers.

**Comment by carrodher** [comment]
Thanks for the ping, I just added my thoughts in one of the inline comments

File: bitnami/common/templates/_labels.tpl
Comment (Starefossen) [reply bitnami/common/templates/_labels.tpl:6]: It looks like Istio is moving more towards the recommended Kubernetes labels so this might not be as needed as I first though ref. https://github.com/istio/istio/issues/10724

**Comment by Starefossen** [comment]
As per discussion https://github.com/istio/istio/issues/10724 is seams that Istio is moving more and more to adopt the now recommended Kubernetes lables and that the old labels are no longer needed.

**Comment by migruiz4** [comment]
Hi @Starefossen ,

In that case, I'd suggest deleting the duplicated labels `app` and `version` from this PR.

Regarding `app.kubernetes.io/version`, I agree with @juan131 and I would recommend being extremely cautious when using labels that could change between chart versions.

I'd like to perform upgrade tests in as many charts as possible before accepting it because `common.labels.standard` is used in almost every template in this repository.

File: bitnami/common/Chart.yaml
Comment (Starefossen) [reply bitnami/common/Chart.yaml:5]: Done ✅

File: bitnami/common/templates/_labels.tpl
Comment (Starefossen) [reply bitnami/common/templates/_labels.tpl]: I have removed the legacy labels.

**Comment by Starefossen** [comment]
Done @migruiz4 ✅

**Comment by migruiz4** [comment]
Hi @Starefossen, @juan131 and @carrodher,

After some investigation, looks like the `common.labels.standard` is only used in the fields `metadata.labels`(All manifests) and `spec.template.metadata.labels`(Statefuls and Deployments).

The only exception I found was:
https://github.com/bitnami/charts/blob/8af317270ab0d665cedad512d9932f4358174770/bitnami/nginx-ingress-controller/templates/controller-poddisruptionbudget.yaml#L15-L24

This will have to be modified before merging this change because `spec.selector` is indeed an [immutable field](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#label-selector-updates), but it seems to be a bug because I think we should instead be using `common.labels.matchLabels`.

So far, upgrade tests changing `metadata.labels` and `spec.template.metadata.labels` were successful, so we can assume they are not immutable.

Additionally, Helm recommends the appVersion field to be quoted so YAML doesn't interpret versions as floating-point values.

File: bitnami/common/templates/_labels.tpl
Comment (migruiz4) [inline bitnami/common/templates/_labels.tpl:9]:
```suggestion
{{- if .Chart.AppVersion }}
app.kubernetes.io/version: {{ .Chart.AppVersion }}
{{- end }}
```

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Please make the changes below, as it is `.Chart.AppVersion` instead of `.Release.appVersion`.

Additionally, the `-}}` at the end will consume the next line, causing issues.

**Pull request closed without merging**
